src/test/java/com/contrastsecurity/ContrastSDKTest.java

```diff
@@ -373,4 +378,54 @@ public void testGetTraceMetadataFilters() {
 
     assertThat(traces.getCount() > 0).isTrue();
   }
+
+  private static Stream<Arguments> make_connection_validate_user_agent_header() {
+    return Stream.of(
+        Arguments.arguments(
+            "sdk with name and version",
+            IntegrationName.INTELLIJ_INTEGRATION,
+            "1.0.0",
+            "INTELLIJ_INTEGRATION/1.0.0"),
+        Arguments.arguments(
+            "sdk with name and without version",
+            IntegrationName.INTELLIJ_INTEGRATION,
+            null,
+            "INTELLIJ_INTEGRATION"),
+        Arguments.arguments("sdk without name and without version", null, null, ""),
+        Arguments.arguments("sdk without name and with version", null, "1.0.0", ""));
+  }
+
+  @ParameterizedTest
+  @MethodSource
+  public void make_connection_validate_user_agent_header(
+      final String testCase,
+      final IntegrationName integrationName,
+      final String version,
+      final String expectedResult)
+      throws IOException {
+
+    final ContrastSDK sdk = buildSdk(integrationName, version);
+
+    URLConnection conn = sdk.makeConnection("https://www.google.com", HttpMethod.GET.toString());
```

Comment:
is this making an actual connection to google?

Reply:
feels wrong to make an actual connection for unit tests.

Reply:
This _shouldn't_ actually make an outgoing connection, because we haven't called any methods on the `HttpURLConnection` that would cause the connection to start. However, we should not connect to google.com for testing. If we want to be sure that we're not sending an outgoing connection in this test, then we should send the connection to a server we know will fail e.g. `does-not-exist.contrastsecurity.com`.

If I'm wrong about this, or if we or the JDK changes `makeConnection` such that it does cause an outgoing request, then we should reconsider the design of this test. We should instead use a mock server that is capable of receiving the request and verifying that we sent the User-Agent header.